File: web/types/enums.ts
export enum AccountType {
  CHECKING = 'checking',
  SAVINGS = 'savings',
  CASH = 'cash',
}

export enum CategoryType {
  INCOME = 'income',
  EXPENSE = 'expense',
}

export enum SyncStatus {
  PENDING = 'pending',
  SYNCED = 'synced',
}

export const AccountTypeLabels: Record<AccountType, string> = {
  [AccountType.CHECKING]: 'Courant',
  [AccountType.SAVINGS]: 'Épargne',
  [AccountType.CASH]: 'Espèces',
}

export const CategoryTypeLabels: Record<CategoryType, string> = {
  [CategoryType.INCOME]: 'Revenus',
  [CategoryType.EXPENSE]: 'Dépenses',
}

File: mobile/lib/src/data/models/enums.dart
/// Types de comptes disponibles
enum AccountType {
  checking, // Compte courant
  savings, // Compte épargne
  cash; // Espèces

  /// Convertit depuis la valeur Supabase (string)
  static AccountType fromString(String value) {
    return AccountType.values.firstWhere(
      (e) => e.name == value,
      orElse: () => AccountType.checking,
    );
  }
}

/// Types de catégories (revenus ou dépenses)
enum CategoryType {
  income,
  expense;

  /// Convertit depuis la valeur Supabase (string)
  static CategoryType fromString(String value) {
    return CategoryType.values.firstWhere(
      (e) => e.name == value,
      orElse: () => CategoryType.expense,
    );
  }
}

/// Statut de synchronisation avec le serveur distant
enum SyncStatus {
  pending, // En attente de synchronisation
  synced; // Synchronisé avec le serveur

  /// Convertit depuis la valeur Supabase (string)
  static SyncStatus fromString(String value) {
    return SyncStatus.values.firstWhere(
      (e) => e.name == value,
      orElse: () => SyncStatus.synced,
    );
  }
}
